resources/lib/tubecast/youtube/app.py

```diff
@@ -430,30 +451,53 @@ def __monitor_volume(self):
 class YoutubeListener(threading.Thread):
 
     def __init__(self, app, ssdp=True):
-        threading.Thread.__init__(self)
-        self.app = app
+        super(YoutubeListener, self).__init__(name="YoutubeListener")
+        self.app = app  # type: YoutubeCastV1
         self.stop = False
         self.ssdp = ssdp
+        self.r = None  # type: Optional[requests.Response]
+
+    def __read_cmd_lines(self, url):  # type: (str) -> Iterator[str]
+        with self.app.session.get(url, stream=True) as self.r:
+            try:
+                for line in self.r.iter_lines():
+                    if self.stop:
+                        break
+
+                    yield line
+            except requests.exceptions.ChunkedEncodingError:
+                # raised when we forcefully close the socket.
+                # If we don't want to stop though, this should raise.
+                if not self.stop:
+                    raise
 
     def _listen(self):
         logger.debug("Listening to youtube remote events...")
         self.app.ofs += 1
-        bind_vals = self.app.bind_vals
+        # copy dict to avoid race condition with __post_bind which sets RID to 1337.
+        # I have no idea whether this is actually a hazard but because this
+        # isn't called very often the performance impact is negligible.
+        bind_vals = self.app.bind_vals.copy()
         bind_vals["RID"] = "rpc"
         bind_vals["CI"] = "0"
         bind_vals["TYPE"] = "xmlhttp"
         bind_vals["AID"] = "3"
-        with self.app.session.get("{}/api/lounge/bc/bind?{}".format(self.app.base_url,
-                                                                    urlencode(bind_vals)), stream=True) as self.r:
-            for line in self.r.iter_lines():
-                self.app.handle_cmd(line)
-        # Restart youtube data input stream if the client is still connected
-        if self.app.has_client and not self.stop:
-            self._listen()
+        url = "{}/api/lounge/bc/bind?{}".format(self.app.base_url, urlencode(bind_vals))
+        for line in self.__read_cmd_lines(url):
+            self.app.handle_cmd(line)
 
     def run(self):
-        if not self.ssdp or (self.app.has_client and not self.stop):
+        while not self.stop and (not self.ssdp or self.app.has_client):
```

Comment:
Instead of using a recursive `_listen` method we now have a while loop to avoid the improbable but technically possible stack overflow.